src/js/components/Tabs/Tabs.js

```diff
@@ -93,37 +185,150 @@ const Tabs = forwardRef(
+                a11yTitle="Previous Tab"
+                icon={<Previous />}
+                disabled={disableLeftArrow}
+                // removed from tabIndex, button is redundant for keyboard users
+                tabIndex={-1}
+                onClick={() => {
+                  let scrolledToIndex;
+                  for (let i = 0; i < tabRefs.length - 1; i += 1) {
+                    if (
+                      !isVisible(tabRefs[i].current) &&
+                      isVisible(tabRefs[i + 1].current)
+                    ) {
+                      if (scroll && scroll.step)
+                        i = Math.max(i - (scroll.step - 1), 0);
+                      scrolledToIndex = i;
+                      tabRefs[i].current.scrollIntoView({ behavior: 'smooth' });
+                      break;
+                    }
+                  }
+
+                  setDisableRightArrow(false);
+                  if (scrolledToIndex === 0) {
+                    // wait for scroll animation to finish
+                    const checkVisible = setInterval(() => {
+                      if (isVisible(tabRefs[0].current)) {
+                        setDisableLeftArrow(true);
+                        clearInterval(checkVisible);
+                      }
+                    }, 100);
+                    setTimeout(() => {
+                      clearInterval(checkVisible);
+                    }, 500);
+                  }
+                }}
+              />
+            )}
+            <StyledTabsHeader
+              ref={headerRef}
+              as={Box}
+              direction="row"
+              justify={justify}
+              alignSelf={alignControls}
+              flex={!!scroll}
+              wrap={!scroll}
+              overflow={scroll ? 'hidden' : 'visible'}
+              background={theme.tabs.header.background}
+              gap={theme.tabs.gap}
+              onBlur={() => setFocusIndex(activeIndex)}
+              {...tabsHeaderStyles}
+            >
+              {tabs}
+            </StyledTabsHeader>
+            {overflow && scroll && (
+              <Button
+                a11yTitle="Next Tab"
+                icon={<Next />}
+                disabled={disableRightArrow}
+                // removed from tabIndex, button is redundant for keyboard users
+                tabIndex={-1}
+                onClick={() => {
+                  let scrolledToIndex = 0;
+                  for (let i = tabRefs.length - 1; i > 0; i -= 1) {
+                    if (
+                      !isVisible(tabRefs[i].current) &&
+                      isVisible(tabRefs[i - 1].current)
+                    ) {
+                      if (scroll && scroll.step) {
+                        i = Math.min(i + (scroll.step - 1), tabRefs.length - 1);
+                      }
+                      scrolledToIndex = i;
+                      tabRefs[i].current.scrollIntoView({ behavior: 'smooth' });
+                      break;
```

Comment:
Is there any way we can implement this logic without using `break` -- I think break is generally discouraged bc it creates multiple ways a loop can be exited

Reply:
That makes sense, I will go back and update this to not use `break`